### PR TITLE
Rosetta Implementation - pt2 (Stage 3.2 of Node API Overhaul)

### PR DESCRIPTION
**Pull request by Daniel-VDM**
# Stage 3.2 of [Node API Overhaul](https://github.com/harmony-one/harmony/issues/3210)

This PR implements Rosetta's [`/network`](https://www.rosetta-api.org/docs/NetworkApi.html) & [`/block`](https://www.rosetta-api.org/docs/BlockApi.html) node endpoint services.

This PR also adds unit tests to the `rosetta` package to ensure correct formatting behavior. 

Basic functionality was tested using a [fork of Rosetta's CLI](https://github.com/Daniel-VDM/rosetta-cli/tree/harmony) (to check transactions). Edge case testing with a testnet node is currently being done (8/22).

Here is the unit test coverage:
```
PASS
coverage: 56.0% of statements
ok  	github.com/harmony-one/harmony/rosetta/services	0.413s
```
> As we don't currently have an easy way to create a mock [`harmony`](https://github.com/harmony-one/harmony/blob/273d0c563331c8f1fea0a93449cb8547b25b2839/hmy/hmy.go#L43) object, this is probably the best we can do without investing a bunch of time to create a testing package (which I think we should do at some point).

### CODE REVIEW

File: node/api.go [hunk 1]
```diff
@@ -86,6 +86,11 @@ func (node *Node) StartRosetta() error {
 	return rosetta.StartServers(harmony, node.NodeConfig.RosettaServer)
 }
 
+// StopRosetta stop rosetta service
```
[the remark this answers is not in the document]
Comment (Daniel-VDM) [reply node/api.go]: Fixed

File: rosetta/services/block.go [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+		return nil, err
+	}
+
+	var block *hmytypes.Block
+	var currBlockID, prevBlockID *types.BlockIdentifier
+	if block, rosettaError = s.getBlock(ctx, request); rosettaError != nil {
+		return nil, rosettaError
+	}
+	currBlockID = &types.BlockIdentifier{
+		Index: block.Number().Int64(),
```
[the remark this answers is not in the document]
Comment (Daniel-VDM) [reply rosetta/services/block.go:55]: Yea it should be, however, the type requires an int64. At some point, rosetta took in -1 to indicate genesis blocks. They have since changed the spec, maybe the SDK will receive an update.

File: rosetta/rosetta.go [hunk 3]
```diff
@@ -59,17 +70,37 @@ func newHTTPServer(handler http.Handler) *http.Server {
...
+					err = errors.New(t)
+				case error:
+					err = t
+				default:
+					err = errors.New("unknown error")
+				}
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				utils.Logger().Error().Err(err).Msg("Rosetta Error")
+				// Print to stdout for quick check of rosetta activity
+				fmt.Printf("%s PANIC: %s\n", time.Now().Format("2006-01-02 15:04:05"), err.Error())
```
[the remark this answers is not in the document]
Comment (Daniel-VDM) [reply rosetta/rosetta.go]: Printed to stderr with a stack trace

**Comment by Daniel-VDM** [comment]
> LGTM. Is rosetta api test part of regression now? also, can we add it to nightly?

We don't have all the endpoints implemented yet for the rosetta integration test. Will add it as part of the PR testing process after implementation, and can add it as a stage to the nighly check afterward.

**Comment by rlan35** [comment]
looks good.
